perception/autoware_shape_estimation/lib/corrector/utils.cpp

```diff
@@ -69,178 +67,99 @@ bool correctWithDefaultValue(
-  }
-  // fit width
-  else if (  // NOLINT
-    (param.min_width < (v_point.at(first_most_distant_index) * 2.0).norm() &&
-     (v_point.at(first_most_distant_index) * 2.0).norm() < param.max_width) &&
-    (param.min_width < (v_point.at(second_most_distant_index) * 2.0).norm() &&
-     (v_point.at(second_most_distant_index) * 2.0).norm() <
-       param.max_width))  // both of edge is within width threshold
-  {                       // NOLINT
-    correction_vector = v_point.at(first_most_distant_index);
-    if (correction_vector.x() == 0.0) {
-      correction_vector.y() =
-        std::max(std::abs(correction_vector.y()), param.default_length / 2.0) *
-          (correction_vector.y() < 0.0 ? -1.0 : 1.0) -
-        correction_vector.y();
-    } else if (correction_vector.y() == 0.0) {
-      correction_vector.x() =
-        std::max(std::abs(correction_vector.x()), param.default_length / 2.0) *
-          (correction_vector.x() < 0.0 ? -1.0 : 1.0) -
-        correction_vector.x();
-    }
-  } else if (  // NOLINT
-    param.min_width < (v_point.at(first_most_distant_index) * 2.0).norm() &&
-    (v_point.at(first_most_distant_index) * 2.0).norm() < param.max_width) {
-    correction_vector = v_point.at(second_most_distant_index);
+  // Helper function to apply correction vector
+  auto applyCorrectionVector = [](Eigen::Vector2d& correction_vector, double default_size) {
     if (correction_vector.x() == 0.0) {
       correction_vector.y() =
-        std::max(std::abs(correction_vector.y()), param.default_length / 2.0) *
+        std::max(std::abs(correction_vector.y()), default_size / 2.0) *
           (correction_vector.y() < 0.0 ? -1.0 : 1.0) -
         correction_vector.y();
     } else if (correction_vector.y() == 0.0) {
       correction_vector.x() =
-        std::max(std::abs(correction_vector.x()), param.default_length / 2.0) *
+        std::max(std::abs(correction_vector.x()), default_size / 2.0) *
           (correction_vector.x() < 0.0 ? -1.0 : 1.0) -
         correction_vector.x();
     }
-  } else if (  // NOLINT
-    param.min_width < (v_point.at(second_most_distant_index) * 2.0).norm() &&
-    (v_point.at(second_most_distant_index) * 2.0).norm() < param.max_width) {
-    correction_vector = v_point.at(first_most_distant_index);
+  };
+
+  // Check if first and second most distant points are on opposite edges (0,2 pair or 1,3 pair)
+  bool are_opposite_edges = (static_cast<int>(std::abs(
+    static_cast<int>(first_most_distant_index) - static_cast<int>(second_most_distant_index))) % 2 == 0);
+  // Calculate distances for clearer condition checking
+  double first_point_distance = (v_point.at(first_most_distant_index) * 2.0).norm();
+  double second_point_distance = (v_point.at(second_most_distant_index) * 2.0).norm();
+  double third_point_distance = (v_point.at(third_most_distant_index) * 2.0).norm();
+  // Check if points are within parameter ranges
+  bool first_in_width_range = (first_point_distance < param.max_width);
```

Comment:
I might missing something, but why we don't need to check the param.min_width for `*_in_width_range` part?  
In the original implementation, it was checking like `param.min_width < (...).norm() && (...).norm() < param.max_width`.